src/NuGet.Core/NuGet.ProjectManagement/NuGetProject.cs

```diff
@@ -35,6 +37,14 @@ protected NuGetProject(Dictionary<string, object> metadata)
             get { return InternalMetadata; }
         }
 
+        public string Guid
```

Comment:
Either way please don't call the member Guid.

If it's a guid make the type guid. If it's opaque make it a string or object. And call it batchid